Assets/Scripts/GranadeerScript.cs

```diff
@@ -17,121 +18,108 @@ enum State
         Die
     }
 
-    State state;
+    [SerializeField] State state;
 
     private GrenadeScript newGrenade = null; 
 
-    private void Awake()
+    private void OnEnable()
     {
         Actions.OnSummonKilled += SummonDestroyed;
     }
 
-    private void OnDestroy()
+    private void OnDisable()
     {
         Actions.OnSummonKilled -= SummonDestroyed;
     }
 
-    private void Start()
+    protected override void Start()
     {
+        base.Start();
         state = State.Chasing;
         target = GameManager.Instance.PlayerTransform.position;
+
+        colliderTrigger.GetList(alliesInRange);
```

Comment:
Move this to the base clases of the summon and enemies,m since all child clases do the same with the list